src/ui/actions/network.ts

```diff
@@ -19,3 +27,15 @@ export const newNetworkRequestsAction = ({
   type: "NEW_NETWORK_REQUESTS",
   payload: { requests, events },
 });
+
+export function fetchFrames(point: string, time: number) {
+  return async ({ dispatch }: { dispatch: AppDispatch }) => {
+    const pause = ThreadFront.ensurePause(point, time);
+    const frames = await pause
+      .getFrames()
+      .then(frames => (frames || []).filter(Boolean))
+      .then(frames => Promise.all(frames.map((frame, i) => createFrame(frame, i))))
+      .then(frames => frames.filter(Boolean));
+    dispatch({ type: "SET_FRAMES", payload: { frames, point } });
```

Comment:
this is a nit, but it's nicer to use awaits...

```js
const pause = await ThreadFront.ensurePause(point, time);
const fetchedFrames = await pause.getFrames()
const  frames = fetchedFrames?.filter(Boolean).map(createFrame)?.filter(Boolean)
dispatch({ type: "SET_FRAMES", payload: { frames, point } });
```

There are some other implicit things here. 
* Not everything is async
* I dont think we need both Boolean filters

Reply:
I think "nicer" is kind of in the eye of the beholder 😄. Personally, I think all of the extra local variables make the await version super noisy, but I'm happy to switch to it if it's closer to the rest of the codebase. 💯 for consistency!

Reply:
fair enough